retail-managment-system/rms-ui/src/app/core/services/loading.interceptor.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { finalize } from 'rxjs/operators'
+import { LoaderService } from './loader.service';
+
+@Injectable()
+export class LoadingInterceptor implements HttpInterceptor {
+
+  private totalRequests = 0
```

Comment:
why use this property?

Reply:
to keep track all requests so we can use the loader any where for any request in the application not only in the pages components and control setting it off or on upon the number of running requests